### PR TITLE
jellyseerr: init at 1.4.1

### PR DESCRIPTION
**Pull request by camillemndn**
###### Description of changes

Initializes new package for jellyseerr. Fixes https://github.com/NixOS/nixpkgs/pull/200599

###### Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [x] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [23.05 Release Notes (or backporting 22.11 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2305-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

**Review by JulienMalka — CHANGES_REQUESTED** [review]
You probably don't have to commit ``package.json`` if it is available upstream.

File: pkgs/servers/jellyseerr/default.nix [hunk 1]
```diff
@@ -0,0 +1,85 @@
+{ lib
+, stdenv
+, fetchFromGitHub
+, makeWrapper
+, mkYarnPackage
+, nodejs
+, nodePackages
+, sqlite
+, pkgs
```
Comment (JulienMalka) [inline pkgs/servers/jellyseerr/default.nix:9]:
```suggestion
```

**Comment by camillemndn** [comment]
@JulienMalka Sorry, I added ```package.json``` again, due to inconsistencies with the update script, and also the CI was failing.

**Comment by tboerger** [comment]
Great work that you got it building correctly!

**Comment by JulienMalka** [comment]
> @JulienMalka Sorry, I added `package.json` again, due to inconsistencies with the update script, and also the CI was failing.

@camillemndn Hmm it's strange I go it building correctly without the ``package.json`` file.

**Comment by JulienMalka** [comment]
Also it looks like there is a problem with ``bcrypt`` on darwin.

**Comment by camillemndn** [comment]
@JulienMalka yes, I get the error:
```cannot build during evaluation because the option 'allow-import-from-derivation' is disabled```

**Comment by JulienMalka** [comment]
> @JulienMalka yes, I get the error: `cannot build during evaluation because the option 'allow-import-from-derivation' is disabled`

Looks like it may be a thing with yarn2nix ? No worries then.

**Comment by camillemndn** [comment]
@pennae Sorry I should have pinged for this one first, my mistake

**Review by pennae — COMMENTED** [review]
checked a few other yarn packages, and this looks very much like many of them. apart from a few nitpicks we'd be inclined to merge as is, unless someone with actual javascript packaging knowledge has more comments.

File: pkgs/servers/jellyseerr/default.nix [hunk 2]
```diff
@@ -0,0 +1,86 @@
...
+      rm -r config/*
+      yarn build
+    )
+    runHook postBuild
+  '';
+
+  postInstall = ''
+    makeWrapper '${nodejs}/bin/node' "$out/bin/jellyseerr" --add-flags \
+        "$out/libexec/jellyseerr/deps/jellyseerr/dist/index.js" \
+        --set NODE_ENV production
```
Comment (SuperSandro2000) [inline pkgs/servers/jellyseerr/default.nix:69]:
```suggestion
    makeWrapper '${nodejs}/bin/node' "$out/bin/jellyseerr" \
      --add-flags "$out/libexec/jellyseerr/deps/jellyseerr/dist/index.js" \
      --set NODE_ENV production
```

File: pkgs/servers/jellyseerr/default.nix [hunk 3]
```diff
@@ -0,0 +1,86 @@
...
+
+  # Fixes "SQLite package has not been found installed" at launch
+  pkgConfig.sqlite3.postInstall = ''
+    export CPPFLAGS="-I${nodejs}/include/node"
+    ${nodePackages.node-pre-gyp}/bin/node-pre-gyp install --prefer-offline --build-from-source --python=${python3}/bin/python --nodedir=${nodejs}/include/node
+  '';
+
+  pkgConfig.bcrypt.postInstall = ''
+    export CPPFLAGS="-I${nodejs}/include/node"
+    ${nodePackages.node-pre-gyp}/bin/node-pre-gyp install --prefer-offline --build-from-source --python=${python3}/bin/python --nodedir=${nodejs}/include/node
```
Comment (SuperSandro2000) [inline pkgs/servers/jellyseerr/default.nix:52]: This isn't cross friendly, right?

File: pkgs/servers/jellyseerr/default.nix [hunk 4]
```diff
@@ -0,0 +1,86 @@
...
+, nodejs
+, nodePackages
+, sqlite
+, fetchYarnDeps
+, python3
+}:
+
+let
+  pin = lib.importJSON ./pin.json;
+in
```
Comment (SuperSandro2000) [inline pkgs/servers/jellyseerr/default.nix:15]: Why not inline this? There are many packages that do this in nixpkgs where commands can be copied.

File: pkgs/servers/jellyseerr/update.sh [hunk 5]
```diff
@@ -0,0 +1,42 @@
+#!/usr/bin/env nix-shell
+#!nix-shell -i bash -p nix curl jq prefetch-yarn-deps nix-prefetch-github
+
+ORG="Fallenbagel"
+PROJ="jellyseerr"
```
Comment (SuperSandro2000) [inline pkgs/servers/jellyseerr/update.sh:5]: Those can be inlined

File: pkgs/servers/jellyseerr/update.sh [hunk 6]
```diff
@@ -0,0 +1,42 @@
...
+
+if [ "$#" -gt 1 ] || [[ "$1" == -* ]]; then
+  echo "Regenerates packaging data for $PROJ."
+  echo "Usage: $0 [git release tag]"
+  exit 1
+fi
+
+tag="$1"
+
+set -euo pipefail
```
Comment (SuperSandro2000) [inline pkgs/servers/jellyseerr/update.sh:15]: This should be the very first line

File: pkgs/servers/jellyseerr/update.sh [hunk 7]
```diff
@@ -0,0 +1,42 @@
+#!/usr/bin/env nix-shell
+#!nix-shell -i bash -p nix curl jq prefetch-yarn-deps nix-prefetch-github
+
+ORG="Fallenbagel"
+PROJ="jellyseerr"
+
+if [ "$#" -gt 1 ] || [[ "$1" == -* ]]; then
```
Comment (SuperSandro2000) [inline pkgs/servers/jellyseerr/update.sh:7]:
```suggestion
if [[ "$#" -gt 1 || "$1" == -* ]]; then
```

File: pkgs/servers/jellyseerr/default.nix (hunk 4 above)
Comment (camillemndn) [reply pkgs/servers/jellyseerr/default.nix:15]: Hi! I don't really understand, what do you recommend?

Comment (camillemndn) [reply pkgs/servers/jellyseerr/default.nix:15]: I guess it is for the commodity of the ```update.sh``` script!

File: pkgs/servers/jellyseerr/default.nix (hunk 3 above)
Comment (camillemndn) [reply pkgs/servers/jellyseerr/default.nix]: This is the fix for the error in https://github.com/NixOS/nixpkgs/pull/200599. Is there a way to make it cross-friendlier?

**Comment by adamcstephens** [comment]
Resolves #176432

File: pkgs/servers/jellyseerr/update.sh [hunk 8]
```diff
@@ -0,0 +1,41 @@
...
+if [[ "$#" -gt 1 || "$1" == -* ]]; then
+  echo "Regenerates packaging data for jellyseerr."
+  echo "Usage: $0 [git release tag]"
+  exit 1
+fi
+
+tag="$1"
+
+set -euo pipefail
+
```
Comment (SuperSandro2000) [inline pkgs/servers/jellyseerr/update.sh:15]:
```suggestion
```

File: pkgs/servers/jellyseerr/default.nix [hunk 9]
```diff
@@ -0,0 +1,91 @@
...
+, glib
+}:
+
+let
+  pin = lib.importJSON ./pin.json;
+in
+
+mkYarnPackage rec {
+  pname = "jellyseerr";
+  version = pin.version;
```
Comment (SuperSandro2000) [inline pkgs/servers/jellyseerr/default.nix:20]:
```suggestion
  inherit (pin) version;
```

**Pull request merged**